src/window.rs

```diff
@@ -56,6 +56,39 @@ struct WindowWrapper {
     fullscreen: bool,
 }
 
+pub fn window_geometry() -> Result<(u64, u64), String> {
+    let prefix = "--geometry=";
+    let arg = std::env::args()
+        .filter(|arg| arg.starts_with(prefix))
+        .next();
+    match arg {
+        Some(arg) => {
+            let input = &arg[prefix.len()..];
+
+            if let [Some(width), Some(height)] = input
+                .split('x')
+                .map(|dimension| dimension.parse::<u64>().ok())
+                .collect::<Vec<_>>()[..]
+            {
+                if width > 0 && height > 0 {
```

Comment:
maybe pull the width and height check into the map function? No reason to encode the operation twice. Further, since they both have the same error, you could use `or_else` to turn the parse error into the invalid format string and the greater than zero error into the invalid geometry string. That way you don't have to have control flow here. Even further, the outer match can be turned into a single map with an `unwrap_or` at the end for INITIAL_DIMENSIONS

Reply:
My goal here isn't to nitpick, but to make the code more concise. I think its important for maintainability and at a glance understandability.

Reply:
No worries. I'm not experienced enough with Rust yet, so I really appreciate your feedback.

Reply:
I removed the outer match, and rewrote the inner logic, but I'm not sure if the `match` block is better this way. I need to make sure there's no `ParseError` when checking for zeros because I want to return a parsing error message on an input like `0xabc`.
What do you think?

Reply:
Maybe something like this: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=75303220a1d75e15ae903e9a51abec9b

Reply:
Yeah, looks good to me. Anything else?